data-serving/data-service/test/controllers/case.test.ts

```diff
@@ -1266,19 +1282,21 @@ describe('PUT', () => {
         expect(res.body.demographics.ageRange.start).toEqual(6);
         expect(res.body.demographics.ageRange.end).toEqual(10);
     });
-    it.skip('update present item should create case revision', async () => {
+    it('update present item should create case revision', async () => {
         const c = new Day0Case(minimalDay0CaseData);
         await c.save();
 
-        const newNotes = 'abc';
+        const newComment = 'abc';
         await request(app)
             .put(`/api/cases/${c._id}`)
-            .send({ ...curatorMetadata, notes: newNotes })
+            .send({ ...curatorMetadata, comment: newComment })
             .expect('Content-Type', /json/)
             .expect(200);
 
         expect(await CaseRevision.collection.countDocuments()).toEqual(1);
-        expect((await CaseRevision.find())[0].case).toMatchObject(c.toObject());
+        expect(
+            JSON.parse(JSON.stringify((await CaseRevision.find())[0].case)),
+        ).toEqual(JSON.parse(JSON.stringify(c.toObject())));
```

Comment:
The 0th revision looks like the new case itself...should this test check for the first revision, i.e. the result of the `PUT` request?

Reply:
In this test case we compare a case before update to the case revision that was crated during the update.
So at the beginning of this test we create a new case `c`. Then we send a request to update the content of this case, thus generating a new case revision based on case `c`. We then compare case `c` with case saved in case revision. The result from PUT request will contain an updated case.

The revision numbering might be a little odd. After creating case gets 0 as revision number, and after an update this number is incremented by 1.